hypervisor/dm/io_req.c

```diff
@@ -616,50 +616,120 @@ void register_pio_emulation_handler(struct acrn_vm *vm, uint32_t pio_idx,
 	vm->emul_pio[pio_idx].io_write = io_write_fn_ptr;
 }
 
+/**
+ * @brief Get an available MMIO node
+ *
+ * This API find an available MMIO node from \p vm.
+ *
+ * @param vm The VM to which the MMIO node is belong to.
+ *
+ * @return If there's an available valid mmio_node return it, otherwise return NULL;
+ */
+static inline struct mem_io_node *get_available_mmio_node(struct acrn_vm *vm)
+{
+	uint16_t idx;
+	struct mem_io_node *mmio_node;
+
+	for (idx = 0U; idx < vm->emul_mmio_regions; idx++) {
+		mmio_node = &(vm->emul_mmio[idx]);
+		if ((mmio_node->range_start == 0UL) && (mmio_node->range_end == 0UL)) {
+			break;
+		}
+	}
+
+	if (idx == vm->emul_mmio_regions) {
+		/* The normal case: get a new one */
+		if (vm->emul_mmio_regions < CONFIG_MAX_EMULATED_MMIO_REGIONS) {
+			(vm->emul_mmio_regions)++;
```

Comment:
Where do we still use emul_mmio_regions?
++ at the time of get_xxx is confusing too.

Reply:
@dongyaozu 
1. hv_emulate_mmio will walk all the registered mmio regions to get the corresponding mmio region emulate handler.
2. Here get_xxx is similar to allocate a resource. If you got a resource, the available resource count should decrease one ?